src/backend/renderer/mod.rs

```diff
@@ -602,6 +602,217 @@ pub fn buffer_type(buffer: &wl_buffer::WlBuffer) -> Option<BufferType> {
+                | drm_fourcc::DrmFourcc::R16
+                | drm_fourcc::DrmFourcc::R8
+                | drm_fourcc::DrmFourcc::Rg1616
+                | drm_fourcc::DrmFourcc::Rg88
+                | drm_fourcc::DrmFourcc::Rgb332
+                | drm_fourcc::DrmFourcc::Rgb565
+                | drm_fourcc::DrmFourcc::Rgb888
+                | drm_fourcc::DrmFourcc::Rgbx1010102
+                | drm_fourcc::DrmFourcc::Rgbx4444
+                | drm_fourcc::DrmFourcc::Rgbx5551
+                | drm_fourcc::DrmFourcc::Rgbx8888
+                | drm_fourcc::DrmFourcc::Rgbx8888_a8
+                | drm_fourcc::DrmFourcc::Uyvy
+                | drm_fourcc::DrmFourcc::Vuy101010
+                | drm_fourcc::DrmFourcc::Vuy888
+                | drm_fourcc::DrmFourcc::Vyuy
+                | drm_fourcc::DrmFourcc::X0l0
+                | drm_fourcc::DrmFourcc::X0l2
+                | drm_fourcc::DrmFourcc::Xbgr1555
+                | drm_fourcc::DrmFourcc::Xbgr16161616f
+                | drm_fourcc::DrmFourcc::Xbgr2101010
+                | drm_fourcc::DrmFourcc::Xbgr4444
+                | drm_fourcc::DrmFourcc::Xbgr8888
+                | drm_fourcc::DrmFourcc::Xbgr8888_a8
+                | drm_fourcc::DrmFourcc::Xrgb1555
+                | drm_fourcc::DrmFourcc::Xrgb16161616f
+                | drm_fourcc::DrmFourcc::Xrgb2101010
+                | drm_fourcc::DrmFourcc::Xrgb4444
+                | drm_fourcc::DrmFourcc::Xrgb8888
+                | drm_fourcc::DrmFourcc::Xrgb8888_a8
+                | drm_fourcc::DrmFourcc::Xvyu12_16161616
+                | drm_fourcc::DrmFourcc::Xvyu16161616
+                | drm_fourcc::DrmFourcc::Xvyu2101010
+                | drm_fourcc::DrmFourcc::Xyuv8888
+                | drm_fourcc::DrmFourcc::Y0l0
+                | drm_fourcc::DrmFourcc::Y0l2
+                | drm_fourcc::DrmFourcc::Y210
+                | drm_fourcc::DrmFourcc::Y212
+                | drm_fourcc::DrmFourcc::Y216
+                | drm_fourcc::DrmFourcc::Y410
+                | drm_fourcc::DrmFourcc::Y412
+                | drm_fourcc::DrmFourcc::Y416
+                | drm_fourcc::DrmFourcc::Yuv410
+                | drm_fourcc::DrmFourcc::Yuv411
+                | drm_fourcc::DrmFourcc::Yuv420
+                | drm_fourcc::DrmFourcc::Yuv420_10bit
+                | drm_fourcc::DrmFourcc::Yuv420_8bit
+                | drm_fourcc::DrmFourcc::Yuv422
+                | drm_fourcc::DrmFourcc::Yuv444
+                | drm_fourcc::DrmFourcc::Yuyv
+                | drm_fourcc::DrmFourcc::Yvu410
+                | drm_fourcc::DrmFourcc::Yvu411
+                | drm_fourcc::DrmFourcc::Yvu420
+                | drm_fourcc::DrmFourcc::Yvu422
+                | drm_fourcc::DrmFourcc::Yvu444
+                | drm_fourcc::DrmFourcc::Yvyu
+        ));
+    }
+
+    if let Ok(format) = crate::wayland::shm::with_buffer_contents(buffer, |_, metadata| metadata.format) {
+        return Some(!matches!(
```

Comment:
And this can maybe be just moved into `src/wayland/shm`

Reply:
Moved